Check & Stop for VM Restore Point

lucas-timeworkstudio opened the pull request:
Patched #231 

Comparing might look weird because I copied from a code space on the base and pasted into my local fork

lucas-timeworkstudio commented:
+ Grouped what the script actually removes should work the same though, Don't know if you want it this way

Raphire commented:
It doesn't seem to properly detect the fact that system restore is enabled on my system. Is this the correct service?

P.S. I found [this reddit thread here](https://www.reddit.com/r/PowerShell/comments/1dxqmnj/system_restore_rabbit_hole/), suggesting it's possible to use the value of the `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SystemRestore` registry key, specifically with the `RPSessionInterval` value. (0 being off, >0 being on)

lucas-timeworkstudio commented:
> It doesn't seem to properly detect the fact that system restore is enabled on my system. Is this the correct service?
> 
> P.S. I found [this reddit thread here](https://www.reddit.com/r/PowerShell/comments/1dxqmnj/system_restore_rabbit_hole/), suggesting it's possible to use the value of the `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SystemRestore` registry key, specifically with the `RPSessionInterval` value. (0 being off, >0 being on)

Let me do some testing on this and see if it works, Ill get back to you tomorrow. I'll be out for the rest of the day

lucas-timeworkstudio commented:
> Sounds good. I've updated the PR to include the latest changes from master. That'll make it easier to evaluate the changes.

Hm so I tried adding some stuff, then tested it and it keeps crashing, no logs, no trace of anything. Im going to try doing a few more things but it might come in finished a bit delayed

Raphire commented:
I've been experimenting with this a bit. Check out my changes here: https://github.com/Raphire/Win11Debloat/compare/master...system-restore

I've made it so a system restore point will always be created in custom mode and removed the prompt. In addition I've added a check for the registry key that indicates whether system restore is enabled. If it isn't it will attempt to enable it. Seems to work pretty well.

lucas-timeworkstudio commented:
> I've been experimenting with this a bit. Check out my changes here: [master...system-restore](https://github.com/Raphire/Win11Debloat/compare/master...system-restore)
> 
> I've made it so a system restore point will always be created in custom mode and removed the prompt. In addition I've added a check for the registry key that indicates whether system restore is enabled. If it isn't it will attempt to enable it. Seems to work pretty well.

If you want to merge your ill close this one

lucas-timeworkstudio commented:
> I've merged it (See #235) Thanks for your help with this :D

Your welcome, thank you for assisting me on that
